### PR TITLE
Fixing type declaration of DynamicFlow

### PR DESCRIPTION
**Pull request by richardbata**
## ✅ Checklist

- [ ] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [ ] Changes are tested and all tests pass
- [ ] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [ ] Changes work in all supported browsers (don't forget IE11)
- [ ] You've updated the documentation if necessary

### CODE REVIEW

File: packages/dynamic-flows/src/flow/DynamicFlow.js [hunk 1]
```diff
@@ -275,7 +307,15 @@ const DynamicFlow = (props) => {
   );
 };
 
-DynamicFlow.propTypes = {
+const DynamicFlowWrapper = (props) => {
+  return (
+    <ErrorBoundary onError={props.onError}>
+      <DynamicFlow {...props} />
+    </ErrorBoundary>
```
Comment (dozortsev) [inline packages/dynamic-flows/src/flow/DynamicFlow.js:314]: I would just use `<ErrorBoundary >` in `return (` of `DynamicFlow`

Comment (dozortsev) [reply packages/dynamic-flows/src/flow/DynamicFlow.js:314]: this `DynamicFlowWrapper` looks redundant and rather problematic 

we should have having this
```js
DynamicFlow.propTypes = DynamicFlowWrapper.propTypes = {
```

Comment (richardbata) [reply packages/dynamic-flows/src/flow/DynamicFlow.js:314]: The issue is it doesn't capture JS errors within the DynamicFlow component...

File: packages/dynamic-flows/src/flow/DynamicFlow.js [hunk 2]
```diff
@@ -275,7 +307,15 @@ const DynamicFlow = (props) => {
...
+  );
+};
+
+DynamicFlow.propTypes = DynamicFlowWrapper.propTypes = {
```
Comment (dozortsev) [inline packages/dynamic-flows/src/flow/DynamicFlow.js:318]: why we need this assignment change?

Comment (richardbata) [reply packages/dynamic-flows/src/flow/DynamicFlow.js:318]: So that the default export (Wrapper) has the same type interface as DynamicFlow component

**Review by JulianG-TW — APPROVED** [review]
Thank you!
It would be nice to get rid of the `injectIntl` hook as well. 
Probably in a future PR.

**Review by JulianG-TW — APPROVED** [review]

**Review by dozortsev — APPROVED** [review]

**Pull request merged**